code/modules/mod/modules/modules_maint.dm

```diff
@@ -20,22 +23,37 @@
 
 /obj/item/mod/module/springlock/on_suit_activation()
 	RegisterSignal(mod.wearer, COMSIG_ATOM_EXPOSE_REAGENTS, PROC_REF(on_wearer_exposed))
+	AddComponent(/datum/component/connect_loc_behalf, mod.wearer, gas_connections)
 
 /obj/item/mod/module/springlock/on_suit_deactivation(deleting = FALSE)
 	UnregisterSignal(mod.wearer, COMSIG_ATOM_EXPOSE_REAGENTS)
+	qdel(GetComponent(/datum/component/connect_loc_behalf))
 
-///Signal fired when wearer is exposed to reagents
-/obj/item/mod/module/springlock/proc/on_wearer_exposed(atom/source, list/reagents, datum/reagents/source_reagents, methods, volume_modifier, show_message)
-	SIGNAL_HANDLER
-
-	if(!(methods & (VAPOR|PATCH|TOUCH)) || set_off || mod.wearer.stat == DEAD)
-		return //remove non-touch reagent exposure
+///Fires a signal to snap shut the springlock when called
+/obj/item/mod/module/springlock/proc/snap_signal()
 	to_chat(mod.wearer, span_danger("[src] makes an ominous click sound..."))
 	playsound(src, 'sound/items/modsuit/springlock.ogg', 75, TRUE)
 	addtimer(CALLBACK(src, PROC_REF(snap_shut)), rand(3 SECONDS, 5 SECONDS))
 	RegisterSignal(mod, COMSIG_MOD_ACTIVATE, PROC_REF(on_activate_spring_block))
 	set_off = TRUE
 
+///Calls snap_signal() when exposed to a reagent via VAPOR, PATCH or TOUCH
+/obj/item/mod/module/springlock/proc/on_wearer_exposed(atom/source, list/reagents, datum/reagents/source_reagents, methods, volume_modifier, show_message)
+	SIGNAL_HANDLER
+
+	if(!(methods & (VAPOR|PATCH|TOUCH)) || set_off || mod.wearer.stat == DEAD)
+		return //remove non-touch reagent exposure
+	snap_signal()
+
+///Calls snap_signal() when exposed to water vapor
+/obj/item/mod/module/springlock/proc/on_wearer_exposed_gas()
+	SIGNAL_HANDLER
+	var/turf/t = loc.loc
+	var/datum/gas_mixture/air = t.return_air()
```

Comment:
1. Don't use single letter vars
2. Don't use loc.loc (do you want `get_turf(src)`?)